src/middlewared/middlewared/plugins/cluster_linux/ctdb_root_dir.py

```diff
@@ -14,44 +28,595 @@ class Config:
+            'forward': True
+        })
+
+        # By this point ctdbd is stopped on all nodes
+        #
+        # The following moves our nodes file and every other
+        # ctdb / cluster-config related file to the new volume
+        # then removes the originals
+        job.set_progress(50, 'Moving system files')
+        dst_uuid = self.__move_ctdb_vol(data)
+
+        # Set the sentinel files that will be removed as each node
+        # processes the SYSTEM_VOL_CHANGE event. Presence of sentinel
+        # means that the cluster node has not finished processing.
+        job.set_progress(60, 'Setting update sentinels')
+        for peer in self.middleware.call_sync('gluster.peer.query'):
+            try:
+                self.middleware.call_sync('gluster.filesystem.create_file', {
+                    'volume_name': data['name'],
+                    'parent_uuid': dst_uuid,
+                    'path': f'.{peer["uuid"]}_UPDATE_IN_PROGRESS'
+                })
             except Exception:
-                # this isn't fatal but still need to log something
-                self.logger.warning('Failed to change permissions on %r', ROOT_DIR_NAME, exc_info=True)
-
-        # change user/group owner to root/root (-1 means leave unchanged)
-        uid = 0 if stat.st_uid != 0 else -1
-        gid = 0 if stat.st_gid != 0 else -1
-        if uid == 0 or gid == 0:
-            dir_fd.fchown(uid, gid)
+                # We're already pretty heavily commited at this point
+                # and so we'll just hope for the best that things shake out
+                # okay.
+                self.logger.warning(
+                    'Failed to generate sentinel for node %s',
+                    peer['hostname'], exc_info=True
+                )
+
+        # The event kicks off backgrounded change
+        job.set_progress(75, 'Sending command to remote nodes to update config')
+        self.middleware.call_sync('gluster.localevents.send', {
+            'event': 'SYSTEM_VOL_CHANGE',
+            'name': data['name'],
+            'uuid': dst_uuid,
+            'forward': True
+        })
+
+        # The cluster update is a backgrounded job, and may take a small
+        # amount of time to complete.
+        remaining = MIGRATION_WAIT_TIME
+        while remaining > 0:
+            contents = self.middleware.call_sync('gluster.filesystem.contents', {
+                'volume_name': data['name'],
+                'uuid': dst_uuid,
+            })
+            wait_for_nodes = []
+            for file_name in contents:
+                if not file_name.endswith('_UPDATE_IN_PROGRESS'):
+                    continue
+
+                peer_uuid = file_name.strip('_UPDATE_IN_PROGRESS')
+                wait_for_nodes.append(peer_uuid[1:])
+
+            if not wait_for_nodes:
+                break
+
+            job.set_progress(80, f'Waiting for peers [{", ".join(wait_for_nodes)}] to finish update.')
+            sleep(10)
```

Comment:
Do we need to wait 10 seconds each iteration or can we check at quicker intervals?

Reply:
I don't mind doing 1 second intervals